src/main/java/com/github/princesslana/eriscasper/rest/guild/GuildChannelCreateRequest.java

```diff
@@ -0,0 +1,39 @@
+package com.github.princesslana.eriscasper.rest.guild;
+
+import com.github.princesslana.eriscasper.data.Snowflake;
+import com.github.princesslana.eriscasper.data.resource.Overwrite;
+import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+/**
+ * @see <a
+ *     href="https://discordapp.com/developers/docs/resources/guild#create-guild-channel-json-params">
+ *     https://discordapp.com/developers/docs/resources/guild#create-guild-channel-json-params</a>
+ */
+@Value.Immutable
+public interface GuildChannelCreateRequest {
+  String getName();
+
+  Optional<Integer> getType();
+
+  Optional<String> getTopic();
+
+  Optional<Integer> getBitrate();
+
+  Optional<Integer> getUserLimit();
+
+  ImmutableList<Overwrite> getPermissionOverwrites();
+
+  Optional<Snowflake> getParentId();
+
+  Optional<Boolean> isNsfw();
+
+  static GuildChannelCreateRequest ofName(String name) {
+    return ImmutableGuildChannelCreateRequest.builder().name(name).build();
+  }
+
+  static GuildChannelCreateRequest ofNameAndCategory(String name, Snowflake category) {
```

Comment:
This is where of will start to be troublesome, cos methods names with And in them are usually bad.

But we'll keep this for now and see how it goes.

Reply:
what makes them bad?